Pessoal, deixei a PR em Draft ainda para analisar o que causou os conflitos no tests:

```
2024-08-06 19:56:30,282 565 ERROR odoo odoo.addons.l10n_br_account_payment_order.tests.test_payment_order_change: ERROR: TestPaymentOrderChange.test_change_not_payment
Traceback (most recent call last):
  File "/__w/l10n-brazil/l10n-brazil/l10n_br_account_payment_order/tests/test_payment_order_change.py", line 141, in test_change_not_payment
    self._payment_order_all_workflow(change_payment_order)
  File "/__w/l10n-brazil/l10n-brazil/l10n_br_account_payment_order/tests/test_base_class.py", line 24, in _payment_order_all_workflow
    payment_order_id.draft2open()
  File "/__w/l10n-brazil/l10n-brazil/l10n_br_account_payment_order/models/account_payment_order.py", line 176, in draft2open
    raise UserError(
odoo.exceptions.UserError: The amount for Partner 'KMEE' is negative or null (0.00) !```

--- a/l10n_br_account_payment_order/models/account_move_line.py
+++ b/l10n_br_account_payment_order/models/account_move_line.py
@@ -163,7 +163,8 @@ class AccountMoveLine(models.Model):
             # Se for uma solicitação de baixa do título é preciso informar o
             # campo debit o codigo original coloca o amount_residual
             if (
-                self.mov_instruction_code_id.id
+                self.payment_mode_id.cnab_write_off_code_id
+                and self.mov_instruction_code_id.id
                 == self.payment_mode_id.cnab_write_off_code_id.id
             ):
                 vals["amount_currency"] = self.credit or self.debit

